### PR TITLE
deps: update tcmalloc from a08f5483 to 4addd0f7

### PR DESCRIPTION
**Pull request by wbpcode**
Commit Message: deps: update tcmalloc from a08f5483 to 4addd0f7
Additional Description:

Update tcmalloc from https://github.com/google/tcmalloc/tree/a08f5483182f41ce626e3f66662966342bb448fa to https://github.com/google/tcmalloc/tree/4addd0f73112338b7425e8dcf744bd23b7bd866f

Pre-work of #21160. Ref it for more background info.

Risk Level: Mid.
Testing: n/a.
Docs Changes: n/a.
Release Notes: n/a.
Platform Specific Features: n/a.

### CODE REVIEW

**Comment by wbpcode** [comment]
cc @mattklein123

**Comment by alyssawilk** [comment]
@phlax any concerns for this one?

**Comment by phlax** [comment]
> any concerns for this one?

no  concerns from me - but i was waiting on a review from @mattklein123 as that was requested

**Comment by wbpcode** [comment]
This is pre-work of #21160. It's OK to waiting for a review from @mattklein123 or merge it directly (If there is no any concerns for other reviewers).

File: test/integration/hotrestart_test.sh
Comment (mattklein123) [inline test/integration/hotrestart_test.sh:156]: Just out of curiosity why is this change needed as part of this PR?

/wait-any

Comment (wbpcode) [reply test/integration/hotrestart_test.sh:156]: The tcmalloc added a new bootstrap log which will pollute the output of ` --hot-restart-version`.

Comment (mattklein123) [reply test/integration/hotrestart_test.sh:156]: Hmm, this seems like a problem, as this will effect other people that have scripts. Are they printing something to stderr directly? I think we need to disable this somehow. What is the log exactly? cc @ckennelly 

/wait

Comment (wbpcode) [reply test/integration/hotrestart_test.sh:156]: Here is the log.
```
$ envoy --hot-restart-version
external/com_github_google_tcmalloc/tcmalloc/internal/percpu_tcmalloc.h:1271] Bytes used per cpu of available 1179936 1572864
11.104
```

Comment (wbpcode) [reply test/integration/hotrestart_test.sh:156]: > Are they printing something to stderr directly?

Yeah. Here are some related code snippets. I don't aware any switch here.

```
// Log function.
void Log(LogMode mode, const char* filename, int line, LogItem a, LogItem b,
         LogItem c, LogItem d, LogItem e, LogItem f) {
  Logger state =
      FormatLog(mode == kLogWithStack, filename, line, a, b, c, d, e, f);
  int msglen = state.p_ - state.buf_;
  (*log_message_writer)(state.buf_, msglen);
}

// Writer.
static void WriteMessage(const char* msg, int length) {
  (void)::write(STDERR_FILENO, msg, length);
}
void (*log_message_writer)(const char* msg, int length) = WriteMessage;
```

Comment (mattklein123) [reply test/integration/hotrestart_test.sh:156]: IMO we have to do a patch here to allow this to be disabled/configured.

Comment (wbpcode) [reply test/integration/hotrestart_test.sh:156]: > IMO we have to do a patch here to allow this to be disabled/configured.

Do we need any log of tcmalloc? If a global switch can be accepted, may be a simple macro is enough. Or need we to enhance the log system of tcmalloc to support multiple log level?

cc @ckennelly

Comment (wbpcode) [reply test/integration/hotrestart_test.sh:156]: > We should just delete that log message. It's not super informative, and not really actionable for users.
> 
> I'll aim to land the change tomorrow to delete it.

Great, thanks.

**Comment by wbpcode** [comment]
cc @mattklein123 friendly ping.

**Review by mattklein123 — APPROVED** [review]
Thanks!

**Pull request merged**
